video-browser-api/src/main/java/no/sikt/nva/vms/browser/VideoBrowserHandler.java

```diff
@@ -4,75 +4,125 @@
                                                                               + " not allowed";
-    private static final String INVALID_QUERY_PARAMETERS_EXCEPTION_MESSAGE = "Invalid offset and/or size values";
     private static final String SIZE = "size";
     private static final String OFFSET = "offset";
     private static final String DEFAULT_SIZE = "10";
     private static final String DEFAULT_OFFSET = "0";
+    private static final String HTTPS_SCHEME = UriWrapper.HTTPS + "://";
+    private final URI apiBaseUrl;
 
-    private final Supplier<VideoProvider> videoProvider;
+    private final SecretsReader secretsReader;
 
-    //    @JacocoGenerated
-    //    public VideoBrowserHandler() {
-    //        this(new Environment(), () -> null);
-    //    }
+    @JacocoGenerated
+    public VideoBrowserHandler() {
+        this(new Environment(), SecretsReader.defaultSecretsManagerClient());
+    }
 
-    public VideoBrowserHandler(final Environment environment, final Supplier<VideoProvider> videoProvider) {
+    public VideoBrowserHandler(final Environment environment, final SecretsManagerClient secretsManagerClient) {
         super(Void.class, environment);
-        this.videoProvider = videoProvider;
+        this.secretsReader = new SecretsReader(secretsManagerClient);
+        this.apiBaseUrl = getBaseUrlFromHost(environment);
     }
 
+    /**TODO: implement Panopto**/
+
     @Override
     protected PagedResult<VideoPresentation> processInput(final Void input, final RequestInfo requestInfo,
                                                           final Context context)
-        throws BadRequestException {
+        throws BadRequestException, UnauthorizedException {
 
-        var pageSize = attempt(
-            () -> Integer.parseInt(requestInfo.getQueryParameterOpt(SIZE).orElse(DEFAULT_SIZE)))
-                           .orElseThrow(exception -> new BadRequestException(INVALID_QUERY_PARAMETERS_EXCEPTION_MESSAGE,
-                                                                             HTTP_BAD_REQUEST));
-
-        var offset = attempt(
-            () -> Integer.parseInt(requestInfo.getQueryParameterOpt(OFFSET).orElse(DEFAULT_OFFSET))).orElseThrow();
+        var pageSize = getPageSize(requestInfo);
+        var offset = getOffset(requestInfo);
 
         if (pageSize < 0 || offset < 0) {
             throw new BadRequestException(NEGATIVE_QUERY_PARAMETERS_EXCEPTION_MESSAGE, HTTP_BAD_REQUEST);
         }
 
-        return videoProvider.get().fetchVideoPresentations(pageSize, offset);
+        var username = requestInfo.getUserName();
+        var institutionFeideDomain = getInstitutionFeideDomain(username);
+        var videoIntegrationConfig = getVideoIntegrationConfig();
+        var institutionConfigKaltura = attempt(
+            () -> getInstitution(videoIntegrationConfig, institutionFeideDomain)).orElseThrow().getKaltura();
+
```

Comment:
getInstitution().orElseThrow().getKaltura()?
burde det ikke være: getInstitution().getKaltura().orElseThrow()?

Reply:
Nei, da det er getInstitution som skal inn i try catch. Splitter denne i to steg, tenkte også at det er litt rart å det sånn, men ville ikke instansiere flere variabler :)